v2/internal/pkg/imagebuilder/rebuild_catalog.go

```diff
@@ -0,0 +1,314 @@
+		if err == nil && buildOptions.Manifest != "" {
+			o.Logger.Info("✅ successfully created catalog")
+			o.Logger.Debug("  manifest list id : %s", id)
+			o.Logger.Debug("  image reference  : %s", ref.String())
+		}
+		if err != nil {
+			return result, excludeCatalogs, err
+		}
+
+		var retries *uint
+		retries = new(uint)
+		*retries = 3
+
+		manifestPushOptions := manifests.PushOptions{
+			Store:                  buildStore,
+			SystemContext:          newSystemContext(o.DestTlsVerify),
+			ImageListSelection:     cp.CopyAllImages,
+			Instances:              nil,
+			RemoveSignatures:       true,
+			SignBy:                 "",
+			ManifestType:           "application/vnd.oci.image.manifest.v1+json",
+			AddCompression:         []string{},
+			ForceCompressionFormat: false,
+			MaxRetries:             retries,
+		}
+
+		destImageRef, err := alltransports.ParseImageName(localDest)
+		if err != nil {
+			return result, excludeCatalogs, err
+		}
+
+		_, list, err := manifests.LoadFromImage(buildStore, id)
+		if err != nil {
+			return result, excludeCatalogs, err
+		}
+
+		o.Logger.Debug("local cache destination (rebuilt-catalog) %s", localDest)
+		o.Logger.Debug("destination image reference %v", destImageRef)
+		o.Logger.Debug("pushing manifest list to remote registry")
+		// push the manifest list to local cache
+		_, digest, err := list.Push(ctx, destImageRef, manifestPushOptions)
+		if err != nil {
+			return result, excludeCatalogs, err
+		}
+
+		_, err = buildStore.DeleteImage(id, true)
+		if err != nil {
+			return result, excludeCatalogs, err
+		}
+
+		o.Logger.Info("✅ successfully pushed catalog manifest list")
+		o.Logger.Debug("  digest           : %s", digest)
+
+		excludeImg := v2alpha1.Image{
+			Name: v.OperatorFilter.Catalog,
+		}
+		excludeCatalogs = append(excludeCatalogs, excludeImg)
+
+		if o.Mode == MirrorToMirror {
+			copyImage := v2alpha1.CopyImageSchema{
+				Origin:      v.OperatorFilter.Catalog,
+				Source:      localDest,
+				Destination: remoteDest,
+				Type:        v2alpha1.TypeOperatorCatalog,
+			}
+			result = append(result, copyImage)
+		}
+	}
+	o.Logger.Info("✅ completed rebuild catalog/s")
+	o.Logger.Debug("result %v", result)
+	return result, excludeCatalogs, nil
+}
+
+func newSystemContext(tlsVerify bool) *types.SystemContext {
+	ctx := &types.SystemContext{
+		RegistriesDirPath:           "",
+		ArchitectureChoice:          "",
+		OSChoice:                    "",
+		VariantChoice:               "",
+		BigFilesTemporaryDir:        "",
+		DockerInsecureSkipTLSVerify: types.NewOptionalBool(!tlsVerify),
+	}
+	return ctx
+}
+
+func getStandardBuildOptions(destination string, tlsVerify bool) (define.BuildOptions, error) {
+	// define platforms
+	platforms := []struct{ OS, Arch, Variant string }{
+		{"linux", "amd64", ""},
+		{"linux", "arm64", ""},
+		{"linux", "ppc64le", ""},
+		{"linux", "s390x", ""},
```

Comment:
Since the archs are already inside of the manifest list, hardcoding it is really needed ?